parser/src/test/scala/hmda/parser/fi/ts/TsDatParserSpec.scala

```diff
@@ -19,34 +21,43 @@ class TsDatParserSpec extends FlatSpec with MustMatchers {
     respondent.zipCode mustBe "22102"
   }
 
-  it should "parse other info about the respondent" in {
+  property("generate proper dat format") {
+    forAll(tsGen) { (ts: TransmittalSheet) =>
+      val parsedTs = TsDatParser(ts.toDAT)
+      val updatedRespondent = parsedTs.respondent.copy(id = parsedTs.respondent.id.replaceFirst("^0+(?!$)", ""))
```

Comment:
I'm having trouble understanding these two lines. It looks like extra symbols were inserted somewhere in the process, and this line is removing them. Why is that?